backend/azure/dtmb/internal/utils/typeconversion.go

```diff
@@ -0,0 +1,315 @@
+package utils
+
+import (
+	"fmt"
+
+	"github.com/microsoft/durabletask-go/internal/protos"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	dtmbprotos "github.com/microsoft/durabletask-go/backend/azure/dtmb/internal/backend/v1"
+)
+
+func ConvertFailureDetails(failureDetails *dtmbprotos.FailureDetails) *protos.TaskFailureDetails {
+	if failureDetails == nil {
+		return nil
+	}
+
+	return &protos.TaskFailureDetails{
+		ErrorType:      failureDetails.GetErrorType(),
+		ErrorMessage:   failureDetails.GetErrorMessage(),
+		StackTrace:     &wrapperspb.StringValue{Value: failureDetails.GetStackTrace()}, // what if this is not string
+		InnerFailure:   ConvertFailureDetails(failureDetails.GetInnerFailure()),
+		IsNonRetriable: !failureDetails.GetRetriable(), // need to negate this
+	}
+}
+
+func ConvertTaskFailureDetails(taskFailureDetails *protos.TaskFailureDetails) *dtmbprotos.FailureDetails {
+	if taskFailureDetails == nil {
+		return nil
+	}
+
+	return &dtmbprotos.FailureDetails{
+		ErrorType:    taskFailureDetails.GetErrorType(),
+		ErrorMessage: taskFailureDetails.GetErrorMessage(),
+		StackTrace:   taskFailureDetails.GetStackTrace().GetValue(),
+		InnerFailure: ConvertTaskFailureDetails(taskFailureDetails.GetInnerFailure()),
+		Retriable:    !taskFailureDetails.GetIsNonRetriable(),
+	}
+}
+
+func ConvertEvent(event *dtmbprotos.Event) (*protos.HistoryEvent, error) {
```

Comment:
NGL this kinda hurts the eyes :)

Please consider splitting it into a method for each history event... such as `convertExecutionCompleted`, etc.